### PR TITLE
Feat/swarinfo 323 send requests from bridge

### PR DESCRIPTION
**Pull request by stlaperriere**
Initiate a request from the bridge. Untill now, the only requests sent were "answers" to inbount requests. Now the user can `queueAndSend` a message to be sent.

### CODE REVIEW

File: src/hive_mind_bridge/include/hive_mind_bridge/InboundResponseHandle.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ifndef HIVE_MIND_BRIDGE_INBOUNDRESPONSEHANDLE_H
+#define HIVE_MIND_BRIDGE_INBOUNDRESPONSEHANDLE_H
+
+#include <hivemind-host/GenericResponseDTO.h>
+#include <string>
+
+class InboundResponseHandle {
```
Comment (cquesnel) [inline src/hive_mind_bridge/include/hive_mind_bridge/InboundResponseHandle.h:7]: Should this class be documented?

File: src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp [hunk 2]
```diff
@@ -80,14 +107,55 @@ void HiveMindBridgeImpl::inboundThread() {
             m_inboundQueue.push(message);
         }
     }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(250));
```
Comment (cquesnel) [inline src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp:111]: Should the sleep values be in a constexpr somewhere?

File: src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp [hunk 3]
```diff
@@ -80,14 +107,55 @@ void HiveMindBridgeImpl::inboundThread() {
...
+                    m_inboundResponsesMap.erase(search);
+
+                    ROS_INFO("RECEIVED VALID RESPONSE");
+                } else {
+                    // Did not receive a response for this request. Was the request sent?
+                    if (handle.getState() == OutboundRequestState::READY) {
+                        m_serializer.serializeToStream(outboundMessage);
+                        handle.setState(OutboundRequestState::SENT);
+                    } else {
+                        // TODO add some retry logic after a timeout
```
Comment (cquesnel) [inline src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp:138]: So right now, if the message is lost (eg we try to send it to a robot that is no longer in the swarm), we will never move on to the next message? Is this intended?

**Review by cquesnel — CHANGES_REQUESTED** [review]

File: src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp (hunk 3 above)
Comment (stlaperriere) [reply src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp:138]: Yes. This situation may arise in another context. Let's say that the bridge sends a Request, but that the response takes some time to come back. We do not want to drop/retry the message immediately. I'll add some timeout logic before dropping the message.

Comment (stlaperriere) [reply src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp:138]: Of course, this logic will only have its use when we define a retry logic.

**Review by cquesnel — APPROVED** [review]

**Pull request merged**